atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraAbsentHostTracker.java

```diff
@@ -59,20 +61,20 @@ public synchronized void shutDown() {
         absentCassandraServers.clear();
     }
 
-    private Set<CassandraServer> cleanupAbsentServer(Set<CassandraServer> absentServersSnapshot) {
+    private ImmutableSet<CassandraServer> cleanupAbsentServer(ImmutableSet<CassandraServer> absentServersSnapshot) {
         absentServersSnapshot.forEach(this::incrementAbsenceCountIfPresent);
         return absentServersSnapshot.stream()
                 .map(this::removeIfAbsenceThresholdReached)
                 .flatMap(Optional::stream)
-                .collect(Collectors.toSet());
+                .collect(ImmutableSet.toImmutableSet());
     }
 
-    private void incrementAbsenceCountIfPresent(CassandraServer cassandraServer) {
+    private synchronized void incrementAbsenceCountIfPresent(CassandraServer cassandraServer) {
```

Comment:
good catch - I think this was deemed safe because it is only accessed by a single thread, but agree that guaranteeing this by adding synchronised is appropriate.